relayer/src/main.rs

```diff
@@ -860,3 +899,46 @@ async fn fetch_merkle_roots(args: FetchMerkleRootsArgs) -> AnyResult<()> {
 
     Ok(())
 }
+
+async fn fetch_historical_proxy_and_checkpoints(
+    mut api_provider: ApiProviderConnection,
+    beacon_client: &BeaconClient,
+    vft_manager_address: ActorId,
+    suri: &str,
+) -> anyhow::Result<(ActorId, ActorId)> {
+    log::info!("Fetching historical proxy address and checkpoint light client address from VFT Manager at {vft_manager_address:#?}");
+    let client = api_provider.gclient_client(suri)?;
+    let remoting = GClientRemoting::new(client);
+    let vft_manager = vft_manager_client::VftManager::new(remoting.clone());
+    let historical_proxy = HistoricalProxy::new(remoting.clone());
+    let eth_events = eth_events_electra_client::EthereumEventClient::new(remoting);
+
+    let historical_proxy_address = vft_manager
+        .historical_proxy_address()
+        .recv(vft_manager_address)
+        .await
+        .map_err(|e| anyhow::anyhow!("Failed to receive historical proxy address: {e:?}"))?;
+    log::info!("Historical proxy address is {historical_proxy_address:#?}");
+    let latest_block = beacon_client
+        .get_block_finalized::<beacon::electra::Block>()
```

Comment:
I think we don't need that. `historical_proxy` can return the tuple list and we may use the one with the greatest slot number.